doc/installing-a-development-environment.md

```diff
@@ -1,5 +1,17 @@
 ## Installing
 
+### Introduction to Installation
+
+The development environment setup requires installation of following softwares:
+
+1. Ruby - Needs to be installed in order to install and use Jekyll. Jekyll helps in tranforming plain texts into static websites.
+
+2. Node.js - Needs to be installed in order to install all the required development and javascript dependencies.
+
+3. Make - Can to be installed in order to build cordova website using Makefile. This installation is optional.
+
+4. Python - Can to be installed in order to invoke simple HTTP server from Makefile. This installation is optional.
```

Comment:
python should probably also be sorted last in all the instructions below then, correct?

Reply:
@janpio  you mean in OS specific installations, python should be last?

Reply:
would make sense to match the order of "software" to the order here, right? especially if it is optional, it should not really be listed before a really required one, right?

Reply:
@janpio  agree with you Jan. Implemented and checked in.